src/scheduler.c

```diff
@@ -74,20 +77,37 @@ void *scheduler_thread(void *args)
   * either file_changed or a condition in the schedule 
   */
         struct timespec tm;
+        int info_period = 3;
     
         if (0 == clock_gettime(CLOCK_REALTIME, &tm) && current_schedule) {
+            static char *current_blocked_macs = NULL;
             char *blocked_macs;
-            time_t unix_time = tm.tv_sec; // ignore?  + (tm.tv_nsec / 1000000000)
+            time_t unix_time = tm.tv_sec; // ignore?  +(tm.tv_nsec / 1000000000)
             blocked_macs = get_blocked_at_time(current_schedule, unix_time);
+             
+            if (NULL == current_blocked_macs) {
+                    current_blocked_macs = strdup(blocked_macs); 
```

Comment:
You don't need to `strdup()` the string - the returned pointer was allocated will malloc & is an independent block of memory.